architecture/2024-08-controller-component-threat-model.md

```diff
@@ -0,0 +1,241 @@
+
+Impact:
+
+1. The attacker could gain unauthorized access to sensitive data including the data it can get from the etcd if proper policies are not in place.
+2. The attacker can manipulate the operations of the cluster, causing unauthorized changes.
+
+Mitigation:
+
+1. **Network Policies**: Implement necessary Network Policies to ensure that communication between components is restricted to authorized components only. For example, the Validating Webhook should only accept requests from the original Kubernetes API Server.
+2. **Strong Certificate Management**: Implement strong certificate management practices, including regular rotation of certificates, using short-lived certificates, and monitoring for certificate anomalies.
+3. **Secure Storage**: Store TLS certificates securely using secrets management solutions to prevent unauthorized access. The certificates and the private keys are being kept in a Secret object on Kubernetes.
+
+#### Interception and manipulation of the communication between the webhook and the Kubernetes API Server
+
+**Risk:**
+
+1. An attacker could intercept and manipulate the communication between the webhook and the Kubernetes API Server. This could lead to unauthorized access, data breaches, or manipulation of admission decisions, potentially compromising the security and integrity of the Kubernetes cluster.
+
+**Mitigation:**
+
+1. **mTLS (Mutual TLS)**: Implement mutual TLS to ensure that both the client (Kubernetes API Server) and the server (Validating Webhook) authenticate each other. This helps in verifying the authenticity of both parties and prevents unauthorized access. Note that mTLS is enabled for the communication between the Validating Webhook and the Kubernetes API Server in Radius as of now.
+2. **Network Policies**: Use Kubernetes Network Policies to restrict the network traffic between the Kubernetes API Server and the webhook. This limits the exposure of the webhook to only trusted sources. Reference: <https://kubernetes.io/docs/concepts/services-networking/network-policies/>.
+
+#### An attacker gaining access to etcd and manually changing resources
+
+Risk: If an attacker gains access to etcd, they can manually change resources stored in the etcd datastore. This could lead to unauthorized access, data breaches, and manipulation of cluster operations.
+
+Mitigation:
+
+1. **Access Controls for etcd**: Implement strict access controls for etcd to ensure that only authorized components can access and modify its data.
+2. **Network and Firewall Policies**: Add the necessary Network and Firewall Policies to filter access to the etcd.
+3. **Audit Logs**
+4. **Encryption**
+
```

Comment:
This is a good example of something that's out of scope 👍 You'll notice it doesn't talk about any of our components. I think we should capture this as an assumption somewhere. Let's discuss in the meeting.